dev/release/post-09-docs.sh

```diff
@@ -42,7 +42,10 @@ pushd "${ARROW_DIR}"
 docker-compose build ubuntu-cpp
 docker-compose build ubuntu-python
 docker-compose build ubuntu-docs
-docker-compose run --rm -v "${ARROW_SITE_DIR}/docs:/build/docs" ubuntu-docs
+docker-compose run --rm \
+  -v "${ARROW_SITE_DIR}/docs:/build/docs" \
+  -e ARROW_DOCS_VERSION="${version}" \
+  ubuntu-docs
```

Comment:
This needs to make sure that API changes don't slip into the docs, though?

Reply:
Added a checkout before the doc generation.